store/store.go

```diff
@@ -33,60 +31,92 @@ The store can be assumed to contain all contiguous blocks between base and heigh
 */
 type BlockStore struct {
 	db dbm.DB
-
-	// mtx guards access to the struct fields listed below it. We rely on the database to enforce
-	// fine-grained concurrency control for its data, and thus this mutex does not apply to
-	// database contents. The only reason for keeping these fields in the struct is that the data
-	// can't efficiently be queried from the database since the key encoding we use is not
-	// lexicographically ordered (see https://github.com/tendermint/tendermint/issues/4567).
-	mtx    tmsync.RWMutex
-	base   int64
-	height int64
 }
 
 // NewBlockStore returns a new BlockStore with the given DB,
 // initialized to the last height that was committed to the DB.
 func NewBlockStore(db dbm.DB) *BlockStore {
-	bs := LoadBlockStoreState(db)
-	return &BlockStore{
-		base:   bs.Base,
-		height: bs.Height,
-		db:     db,
-	}
+	return &BlockStore{db}
 }
 
 // Base returns the first known contiguous block height, or 0 for empty block stores.
 func (bs *BlockStore) Base() int64 {
```

Comment:
I've noticed that the current design pattern with this store is to suppress any errors from arising by either panicking or failing silently. We don't even have any logging with the block store (as opposed to say the evidence pool). Are we fine with continuing with this? In contrast the state store bubbles up errors for almost every method.

Reply:
We should return errors. But that will probably have ripple-effects to many other parts of the code, so it's better to change that in a separate PR to avoid too much noise (and since it's a logically separate change).